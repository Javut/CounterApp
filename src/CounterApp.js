import React, {useState}  from 'react'
import PropTypes from 'prop-types';




const CounterApp = ({value}) => {

    const [counter, setCounter] = useState(value); 

    console.log(counter );


    const handleAdd = () => {setCounter(counter+1)}

    const handleLess = () => {setCounter(counter-1)}

    const handleRestart = () => {setCounter(value)}
    

    return (
        <>
            <h1>CounterApp</h1>
            <h2>{counter}</h2>

            <button onClick={handleAdd}>Incrementar Estado</button>
            <button onClick={handleRestart}>Reiniciar Estado </button>
            <button onClick={handleLess}>Decrementar Estado </button>
        </>


      );
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}

export default CounterApp;