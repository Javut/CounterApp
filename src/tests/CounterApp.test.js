import React from 'react';
import {render} from '@testing-library/react';
import '@testing-library/jest-dom';
import CounterApp from '../CounterApp';
import { shallow } from 'enzyme';

describe('Pruebas en el  <CounterApp />', () => { 

    let wrapper;

    beforeEach(() => {
        
        const value = 10;
        wrapper = shallow(<CounterApp value={value}/>);
    })

    

    test('debe de mostrar <CounterApp /> correctamente ', () => { 

        
        expect(wrapper).toMatchSnapshot();

     }); 


     test('debe de mostrar el valor por defecto de 100', () => { 

        const value = 100;
        const wrapper = shallow(<CounterApp  value = {value}
            />
            );

            const valueForDefault = wrapper.find('h2').text().trim();
            expect(valueForDefault).toBe('100');

      })

      test('debe incrementar con el boton +1', () => { 

        // const value = 10;
        // const wrapper = shallow(<CounterApp value={value}/>);
        wrapper.find('button').at(0).simulate('click');
        const valueCounter = wrapper.find('h2').text().trim();
        console.log("Aqui va el counterText" + valueCounter);
        expect(valueCounter).toBe('11');


      })

      test('debe decrementar con el boton -1', () => { 

        // const value = 10;
        // const wrapper = shallow(<CounterApp value={value}/>);
        wrapper.find('button').at(2).simulate('click');
        const valueCounter = wrapper.find('h2').text().trim();
        console.log("Aqui va el counterText" + valueCounter);
        expect(valueCounter).toBe('9');


      })

      test('Prueba del boton reset', () => { 

        // const value = 10;
        // const wrapper = shallow(<CounterApp value={value}/>);
        wrapper.find('button').at(2).simulate('click');
        wrapper.find('button').at(2).simulate('click');
        wrapper.find('button').at(1).simulate('click');
        const valueCounter = wrapper.find('h2').text().trim();
        console.log("Aqui va el counterText" + valueCounter);
        expect(valueCounter).toBe('10');


      })



})
