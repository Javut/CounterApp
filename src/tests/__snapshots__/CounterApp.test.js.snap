// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el  <CounterApp /> debe de mostrar <CounterApp /> correctamente  1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
    Incrementar Estado
  </button>
  <button
    onClick={[Function]}
  >
    Reiniciar Estado 
  </button>
  <button
    onClick={[Function]}
  >
    Decrementar Estado 
  </button>
</Fragment>
`;
